server/src/auth/token_validator.rs

```diff
@@ -75,23 +82,27 @@ impl TokenValidator {
                     }
                 })
                 .collect();
-            self.edge_sink.sink_tokens(tokens_to_sink.clone()).await?;
-            Ok(tokens_to_sink.merge(known_tokens))
+            tokens_to_sink.iter().for_each(|t| {
+                self.token_cache.insert(t.token.clone(), t.clone());
```

Comment:
Not currently, no. Needs a bit of thought, because if we expire tokens and you lose access to Unleash, you'll lose access to Edge as well, which is a bad resilience story. But we shouldn't keep them around forever either